oak/server/storage/fixed_nonce_generator.h

```diff
@@ -33,8 +35,18 @@ class FixedNonceGenerator : public asylo::NonceGenerator<kAesGcmSivNonceSize> {
   // Called by asylo::AesGcmSiv::Seal.
   asylo::Status NextNonce(const std::vector<uint8_t>& key_id, AesGcmSivNonce* nonce) override {
     CHECK(nonce != nullptr);
-    std::copy(key_id.begin(), key_id.end(), nonce->begin());
-    std::copy(datum_name_.begin(), datum_name_.end(), nonce->begin() + key_id.size());
+
+    SHA256_CTX context;
+    SHA256_Init(&context);
+    SHA256_Update(&context, datum_name_.data(), datum_name_.size());
+    SHA256_Update(&context, key_id.data(), key_id.size());
+
+    std::vector<uint8_t> digest;
```

Comment:
Might be over-optimization, but it feels like this would be better using a 32B fixed size array and then just copy the first 12B of it.

Reply:
Done.